src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -1514,6 +1506,19 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
 
   protected def getScope(flags: JavaAccFlags): Scope =
     if (flags.isStatic) staticScope else instanceScope
+
+  // Preserve order, but discard a previous matching attibute and keep the last, since runtime attrs include args.
+  private def addUniqueAnnotation(symbol: Symbol, annot: AnnotationInfo): symbol.type = {
+    def addIt(annots: List[AnnotationInfo], rest: List[AnnotationInfo]): List[AnnotationInfo] =
+      rest match {
+        case h :: t =>
+          val loopy = addIt(annots, t)
+          if (h.atp.typeSymbol.initialize.isJavaDefined && h.matches(annot.atp.typeSymbol.initialize)) loopy
+          else h :: loopy
+        case Nil => annots
```

Comment:
```suggestion
        case Nil => List(annot)
```
then we don't need `annots`

Reply:
That was my original version. I tired of editing back and forth, so I made it a fold just to keep my interest. But I decided not to rely on "order" (whether Deprecated attribute is seen before the RuntimeVisible annotation), so for simplicity it probes for deprecation first.